### PR TITLE
Rename DA.Private.TextMap into DA.TextMap

### PR DESCRIPTION
**Pull request by hurryabit**
Also clean up the code a bit.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/573)

### CODE REVIEW

File: daml-foundations/daml-ghc/daml-stdlib-src/DA/TextMap.daml [hunk 1]
```diff
@@ -74,15 +73,14 @@ insert = primitive @"BEMapInsert"
...
 
-
-instance (Eq a) => Eq (TextMap a) where
+instance Eq a => Eq (TextMap a) where
   x == y = toList x == toList y
 
-instance (Ord a) => Ord (TextMap a) where
-  compare = \m1 m2 -> compare (toList m1) (toList m2)
+instance Ord a => Ord (TextMap a) where
+  x `compare` y = toList x `compare` toList y
```
Comment (cocreature) [inline daml-foundations/daml-ghc/daml-stdlib-src/DA/TextMap.daml]: We should add `comparing` to the stdlib.

Comment (hurryabit) [reply daml-foundations/daml-ghc/daml-stdlib-src/DA/TextMap.daml]: The question is where to put it. `DA.Ord`? Or do we want `on`? In `DA.Prelude` or `DA.Function`?

Comment (cocreature) [reply daml-foundations/daml-ghc/daml-stdlib-src/DA/TextMap.daml]: `DA.Ord` sounds sensible.

Comment (neil-da) [reply daml-foundations/daml-ghc/daml-stdlib-src/DA/TextMap.daml]: I've not found `comparing` to be that valuable in Haskell - I actually removed the HLint hints pointing people at it.

**Pull request merged**